File: src/main/java/me/tomasan7/jecnadesktop/repository/TimetableRepository.java
package me.tomasan7.jecnadesktop.repository;

import me.tomasan7.jecnadesktop.data.Timetable;

import java.util.concurrent.CompletableFuture;

/**
 * Retrieves {@link Timetable} from any kind of data source.
 */
public interface TimetableRepository
{
	Timetable queryTimetable ();

	default CompletableFuture<Timetable> queryTimetableAsync ()
	{
		return CompletableFuture.supplyAsync(this::queryTimetable);
	}
}

File: src/main/kotlin/me/tomasan7/jecnadesktop/repository/TimetableRepository.kt
package me.tomasan7.jecnadesktop.repository

import me.tomasan7.jecnadesktop.data.SchoolYear
import me.tomasan7.jecnadesktop.data.Timetable

/**
 * Retrieves [Timetable] from any kind of data source.
 */
interface TimetableRepository
{
    suspend fun queryTimetable(): Timetable

    /**
     * @param schoolYear The [SchoolYear] to get the [Timetable] for.
     */
    suspend fun queryTimetable(schoolYear: SchoolYear): Timetable
}